File: algorithms/math/python/gcd.py
import math


def gcd(a: int, b: int) -> int:
    return math.gcd(a, b)


def _manual_gcd(a: int, b: int) -> int:
    absA = abs(a)
    absB = abs(b)
    while absB:
        absA, absB = absB, absA % absB
    return absA

File: algorithms/math/javascript/gcd.js
function gcd(a, b) {
  if (!Number.isInteger(a) || !Number.isInteger(b)) {
    throw new TypeError(
      `gcd() arguments must be integers, got: a=${typeof a}, b=${typeof b}`
    );
  }
  let absA = Math.abs(a);
  let absB = Math.abs(b);
  while (absB !== 0) {
    [absA, absB] = [absB, absA % absB];
  }
  return absA;
}
